libraries/libc/stdio/sscanf.cpp

```diff
@@ -0,0 +1,528 @@
+#include <assert.h>
+#include <ctype.h>
+#include <libutils/Scanner.h>
+#include <stdio.h>
+#include <string.h>
+
+namespace
+{
+enum
+{
+    SCANF_TYPE_CHAR,
+    SCANF_TYPE_SHORT,
+    SCANF_TYPE_INTMAX,
+    SCANF_TYPE_L,
+    SCANF_TYPE_LL,
+    SCANF_TYPE_PTRDIFF,
+    SCANF_TYPE_SIZE_T,
+    SCANF_TYPE_INT
+};
+}
+
+static void store_int(void *dest, unsigned int size, unsigned long long i)
+{
+    switch (size)
+    {
+    case SCANF_TYPE_CHAR:
+        *(char *)dest = i;
+        break;
+    case SCANF_TYPE_SHORT:
+        *(short *)dest = i;
+        break;
+    case SCANF_TYPE_INTMAX:
+        *(intmax_t *)dest = i;
+        break;
+    case SCANF_TYPE_L:
+        *(long *)dest = i;
+        break;
+    case SCANF_TYPE_LL:
+        *(long long *)dest = i;
+        break;
+    case SCANF_TYPE_PTRDIFF:
+        *(ptrdiff_t *)dest = i;
+        break;
+    case SCANF_TYPE_SIZE_T:
+        *(size_t *)dest = i;
+        break;
+    /* fallthrough */
+    case SCANF_TYPE_INT:
+    default:
+        *(int *)dest = i;
+        break;
+    }
+}
+
+template <typename H>
+static int do_scanf(H &handler, Scanner &scan, va_list args)
```

Comment:
I think handler can also be a scanner. This might require to use the one from new-io tho.

Reply:
Can't use that yet in **libc** due to it being in **libsystem**. Correct me if i'm wrong